### PR TITLE
added pypi-ci

### PR DESCRIPTION
**Pull request by CarlottaSartore**
This should address #35 

Now on releases, urdf-modifier should be  release on pypi

### CODE REVIEW

**Comment by CarlottaSartore** [comment]
C.C. @Nicogene @traversaro  @mfussi66

**Comment by traversaro** [comment]
I guess we need to add a token secret here. I asked back in time to PyPI to get an organization, but I need to check what happen to that request. In the meanwhile I guess we need to add a personal token. @Nicogene @mfussi66 if you want I can handle that, let me know if that is ok.

**Comment by Nicogene** [comment]
> I guess we need to add a personal token. @Nicogene @mfussi66 if you want I can handle that, let me know if that is ok.

Yes sure!

**Comment by mfussi66** [comment]
> I guess we need to add a token secret here. I asked back in time to PyPI to get an organization, but I need to check what happen to that request. In the meanwhile I guess we need to add a personal token. @Nicogene @mfussi66 if you want I can handle that, let me know if that is ok.

No problem!

**Comment by traversaro** [comment]
I added the tokens, the PR is ready for review. @Nicogene @CarlottaSartore @mfussi66 please give me your pypi handles to be added to the project, thanks!

**Comment by Nicogene** [comment]
> I added the tokens, the PR is ready for review. @Nicogene @CarlottaSartore @mfussi66 please give me your pypi handles to be added to the project, thanks!

Mine is `Nicogene`

**Comment by traversaro** [comment]
> `Nicogene`

Ok, I will add you as soon as the first release exists, otherwise before there is no pypi project to add you to.

File: README.md [hunk 1]
```diff
@@ -39,11 +39,11 @@ Install `python3`, if not installed (in **Ubuntu 20.04**):
 sudo apt install python3.8
 ```
 
-Clone the repo and install the library:
+ Install the library via pip:
 
 ```bash
 
-pip install "urdfModifiers @ git+https://github.com/icub-tech-iit/urdf-modifiers"
+pip install urdfModifiers
```
Comment (traversaro) [inline README.md]:
```suggestion
pip install urdf-modifiers 
```

**Review by traversaro — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (traversaro) [reply README.md]: I would also change https://github.com/CarlottaSartore/urdf-modifiers/blob/adding_pypi_ci/setup.cfg#L2 .

Comment (traversaro) [reply README.md]: Argh, but the python module name is `urdfModifiers` .

Comment (traversaro) [reply README.md]: To clarify why I do not like `urdfModifiers`, camel case is not suggested in PEP8 guidelines, see https://peps.python.org/pep-0008/#package-and-module-names, and not so common at all (see https://pypistats.org/top). Anyhow, I don't want to create more friction than necessary, so if you want to go forward with `urdfModifiers`, feel free to proceed.

Comment (traversaro) [reply README.md]: Thinking more about this, I think we should at least call the package name `urdf-modifiers` . We can always change the module name later from `urdfModifiers` to `urdf-modifiers`  in a backward compatible way, but changing python package name would be complex.

**Comment by mfussi66** [comment]
> I added the tokens, the PR is ready for review. @Nicogene @CarlottaSartore @mfussi66 please give me your pypi handles to be added to the project, thanks!

Mine is `mfuzz`.

File: README.md (hunk 1 above)
Comment (CarlottaSartore) [reply README.md]: Done in 71f49ee9c18b9b74a9afd9673cd29e06eca83d60 and  8747f1591c7b9f3425dca7cffcdbf5c985122d5e

**Comment by CarlottaSartore** [comment]
> I added the tokens, the PR is ready for review. @Nicogene @CarlottaSartore @mfussi66 please give me your pypi handles to be added to the project, thanks!

Mine is `CarlottaSartore`

File: setup.cfg [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [metadata]
-name = urdfModifiers
+name = urdf-modifiers 
 description = library to modify urdf 
 author = "TODO"
 author_email = TODO
```
Comment (traversaro) [inline setup.cfg:5]:
```suggestion
author = "Germán Rodriguez,  Alexandre Antunes"
author_email = "grmanrod@gmail.com"
mantainer = "Nicolo Genesio, Mattia Fussi"
mantainer_email = "nicolo.genesio@iit.it, mattia.fussi@iit.it"
```

Comment (traversaro) [reply setup.cfg:5]: cc @Nicogene @mfussi66 please double check, thanks!

fyi @CarlottaSartore

File: setup.cfg [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [metadata]
-name = urdfModifiers
+name = urdf-modifiers
```
Comment (traversaro) [inline setup.cfg]:
```suggestion
name = urdf-modifiers
```

trailing whitespace

Comment (CarlottaSartore) [reply setup.cfg]: Thanks done

File: setup.cfg (hunk 2 above)
Comment (Nicogene) [reply setup.cfg:5]: Ok for mine, I am not sure about the German email :D

Comment (traversaro) [reply setup.cfg:5]: > Ok for mine, I am not sure about the German email :D

I got it from LinkedIn.

**Review by traversaro — APPROVED** [review]

**Comment by traversaro** [comment]
Thanks @Nicogene ! Can we publish a new release to check if the PyPI upload is working fine?

**Comment by Nicogene** [comment]
Done here:
- https://github.com/icub-tech-iit/urdf-modifiers/releases/tag/v0.0.2

**Comment by Nicogene** [comment]
Unfortunately it failed (cc @traversaro @CarlottaSartore)

- https://github.com/icub-tech-iit/urdf-modifiers/actions/runs/6848316286/job/18618273767

**Pull request merged**